### ISSUE TITLE
Solution on how to work around the issue with logging mlflow artifacts with s3.

### ISSUE THREAD

**Issue by RogerioLS** [ultralytics/ultralytics#18553]
### Search before asking

- [X] I have searched the Ultralytics [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar feature requests.

### Description

Hello everyone, I recently came across this problem where MLflow artifacts were not registered in S3 even though I provided the artifact address. I saw that some colleagues were also experiencing this, so I decided to share the code that helped me solve this problem. It is very simple.
Here is the code.
```python
import mlflow
import mlflow.pytorch
from ultralytics import YOLO
import os
import time

os.environ['WANDB_DISABLED'] = 'true'

# Configurações do AWS
os.environ['AWS_ACCESS_KEY_ID'] = ""
os.environ['AWS_SECRET_ACCESS_KEY'] = ""
os.environ['AWS_DEFAULT_REGION'] = ''

EXPERIMENT_NAME = "YOLOv8 Full Parameter Tracking"
os.environ["MLFLOW_TRACKING_URI"] = "http://ec2-"
mlflow.set_experiment(EXPERIMENT_NAME)

experiment = mlflow.get_experiment_by_name(EXPERIMENT_NAME)
print("experiment_id:", experiment.experiment_id)
print("artifact_location:", experiment.artifact_location)
print("lifecycle_stage:", experiment.lifecycle_stage)

try:
    with mlflow.start_run(experiment_id=experiment.experiment_id, run_name="") as run:
        model = YOLO('yolov8x.pt')

        # Define as tags para o experimento
        tags = {"Model": "Yolov8", "User": "",
                "TCC": "", "Approach": "best_model"}
        mlflow.set_tags(tags)

        # Loga o modelo treinado
        mlflow.pytorch.log_model(model, "model")

        # Treina o modelo YOLOv8
        results = model.train(
            data='',  # Caminho do arquivo YAML do dataset
            epochs=100,
            batch=16,
            imgsz=640
        )

        # Valida o modelo
        val_results = model.val('')

except mlflow.exceptions.MlflowException as e:
    print(f"Erro no MLflow: {str(e)}")
except Exception as e:
    print(f"Erro inesperado: {str(e)}")

print("Training completed and fully tracked with MLflow.")
```
Another detail is that for my architecture I am using EC2 for tracking.
I hope this helps, 

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @RogerioLS, thank you for sharing this detailed example and for contributing to the Ultralytics community 🚀!

We encourage you to explore the [Ultralytics Docs](https://docs.ultralytics.com) for additional ways to improve your workflow, including [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage guides covering various scenarios.

If this is a 🐛 Bug Report related to MLflow logging and S3, providing a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/) or additional details, such as specific error messages, configurations, or logs, can help us further investigate.

For users looking to integrate third-party tools like MLflow or AWS with Ultralytics models, we recommend confirming that your **Python version (>=3.8)**, **PyTorch version (>=1.8)**, and all other dependencies are up-to-date. You can ensure this by upgrading your `ultralytics` package:

```bash
pip install -U ultralytics
```

### Quickstart Resources

YOLO models can be run in multiple verified environments that include all preinstalled dependencies for smooth operation:
- **Notebooks**: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Cloud Platforms**: Refer to our [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/) or [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/) for deeper cloud integration.
- **Docker Image**: Easily deploy YOLO with our pre-configured [Docker](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) environment <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>.

### Community Support

If you'd like to discuss this further or require real-time assistance, join our growing community:
- [Discord](https://discord.com/invite/ultralytics) 🎧 for quick conversations.
- [Discourse](https://community.ultralytics.com) for in-depth discussions.
- [Subreddit](https://reddit.com/r/Ultralytics) to share tips and experiences.

### CI Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, the [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule) is passing, indicating the robustness of all YOLO modes and tasks on macOS, Windows, and Ubuntu.

This is an automated response 📡, but rest assured, an Ultralytics engineer will review your issue and respond soon if needed. Thank you for your contribution! 🙌

**Comment by visionNoob** [comment]
Awesome code! 
I did hate to use mlflow with S3 back-end when i train a model with Ultralytics. 
Definitely, they said, it support many third-parties, but many of these are not perfect yet - thank you for this.
By the way, if do it this way, does the logs (like train/val loss, ...) get plotted in real-time on MLflow during training?

**Comment by glenn-jocher** [comment]
@visionNoob thank you for your feedback! Yes, when integrating MLflow with Ultralytics YOLO, training and validation metrics, including losses, are logged and should appear in real-time during training, depending on your MLflow setup. For more details, you can refer to the [MLflow integration documentation](https://docs.ultralytics.com/integrations/mlflow/).

**Comment by RogerioLS** [comment]
@glenn-jocher, what an honor to have you here. lol.
I thank you for accepting my solution in such a friendly way. I've been working with Yolo since 2022 and I always recommend it in projects wherever I go. And if I find anything, I'll request a PR. Good luck to you and count on me to further promote this amazing work that Ultralytics offers.
@visionNoob Exactly what @glenn-jocher said, as the epochs are being trained, you can follow in real time.

**Comment by glenn-jocher** [comment]
Thank you for your kind words and continued support of YOLO! It's great to see the community contributing and sharing solutions. If you encounter any other insights or improvements, feel free to submit a PR—we truly value community involvement in advancing Ultralytics YOLO.
